lib/services/devices/deviceService.js

```diff
@@ -104,6 +106,75 @@ function createInitialEntityHandler(deviceData, newDevice, callback) {
     };
 }
 
+/**
+ * Creates the response handler for the initial entity creation request using NGSIv2.
+ * This handler basically deals with the errors that could have been rised during
+ * the communication with the Context Broker.
+ *
+ * @param {Object} deviceData       Object containing all the deviceData needed to send the registration.
+ * @param {Object} newDevice        Device object that will be stored in the database.
+ * @return {function}               Handler to pass to the request() function.
+ */
+function createInitialEntityHandlerNgsi2(deviceData, newDevice, callback) {
```

Comment:
Maybe existing createInitialEntityHandler() should be renamed to createInitialEntityHandlerNgsi1() ?

Reply:
Fixed with 056cba5b7d5a66a78a749654cbc737fb90120f0f